tests/publiccloud/migration.pm

```diff
@@ -0,0 +1,56 @@
+ # SUSE's openQA tests
+ #
+ # Copyright 2016-2022 SUSE LLC
+ # SPDX-License-Identifier: FSFAP
+
+ # Summary: Basic DMS test
+ #
+ #   This test does the following
+ #    - Installs SLE15-Migration and sle15-activation
+ #    - Reboot the system
+ #    - Check the system version is the expected one
+ #
+ # Maintainer: Jesus Bermudez Velazquez <jesus.bv@suse.com>
+
+ use Mojo::Base 'publiccloud::basetest';
+ use testapi;
+ use strict;
+ use warnings;
+ use utils;
+ use publiccloud::utils;
+
+ our $target_version = get_required_var('TARGET_VERSION');
+
+ sub run {
+     my ($self, $args) = @_;
+     $self->select_serial_terminal;
+     my $provider = $self->provider_factory();
+     my $instance = $provider->create_instance();
+     $instance->wait_for_guestregister();
+
+     record_info('INFO', $target_version);
+     record_info('INFO', get_var('TARGET_VERSION'));
+
+     sleep 90; # wait for a bit for zypper to be available
+
+     record_info('installs SLE15-Migration and suse-migration-sle15-activation');
+
```

Comment:
include zypper up and zypper ref before installing package.

Reply:
I think `zypper install` already does a `zypper ref` before running the installation itself
wrt `zypper up` I think there should be a test that runs the migration without `zypper up` 
as it is a use case more than likely customer will do

Reply:
this is positive base test case. Before migration , we expect customers to be latest patches before migration.